src/services/coingecko.ts

```diff
@@ -62,32 +65,48 @@ interface Price {
   price24HChange: number;
 }
 
+const nativeAssetIdMap = {
+  1: 'ethereum',
+  42: 'ethereum',
+  137: 'matic-network'
+};
+
+const platformIdMap = {
+  1: 'ethereum',
+  42: 'ethereum',
+  137: 'polygon-pos'
```

Comment:
How about if we have a function to resolve these mappings?

`function resolveNativeId(networkId: string | number) { ... }` // returns `ethereum`, `matic-network`, etc... (with a fallback to `ethereum` if the network id does not match)

`function resolvePlatformId(networkId: string | number) { ... }` // returns `ethereum`, `polygon-pos`, etc.. (with a fallback to `ethereum` if the network id does not match)

This way, if we are on Rinkeby it wouldn't fail.